src/Plugin/Deriver/EntityIndex.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\services\Plugin\Deriver\EntityIndex.php
+ */
+
+namespace Drupal\services\Plugin\Deriver;
+
+
+use Drupal\ctools\Plugin\Deriver\EntityDeriverBase;
+
+class EntityIndex extends EntityDeriverBase {
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    foreach ($this->entityManager->getDefinitions() as $entity_type_id => $entity_type) {
+      $this->derivatives[$entity_type_id] = $base_plugin_definition;
+      $this->derivatives[$entity_type_id]['title'] = $this->t('@label: Index', ['@label' => $entity_type->getLabel()]);
```

Comment:
Are we safe to assume all entities will have a "getLabel()" or is this just best practice?